File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.Arrays;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        Arrays.sort(phone_book);
        
        for(int i = 0; i < phone_book.length; i++) {
        	try {
               	if(phone_book[i + 1].startsWith(phone_book[i])) {
            		answer = false;
            		break;
            	}
        	} catch(Exception e) {}
 
        }
        

        
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    sorted_phone_book : list[str] = sorted(phone_book)

    size = len(phone_book) - 1

    for i in range(0, size):
        if sorted_phone_book[i + 1].startswith(sorted_phone_book[i]):
            return False

    return True